src/cargo/ops/cargo_install.rs

```diff
@@ -189,6 +195,18 @@ impl<'gctx> InstallablePackage<'gctx> {
                 pkg.to_string()
             ))?;
         }
+        // When --lockfile-path is set, check that passed lock file exists
+        // (unlike the usual flag behavior, lockfile won't be created as we imply --locked)
+        if gctx.locked() {
+            if let Some(requested_lockfile_path) = ws.requested_lockfile_path() {
+                if !ws.root().join("Cargo.lock").exists() {
+                    bail!(
+                        "no Cargo.lock file found in {}",
+                        requested_lockfile_path.display()
+                    );
+                }
```

Comment:
We are going to use the alternative lockfile, so don't need to check lockfile in the workspace root I believe.

```suggestion
                bail!(
                    "no Cargo.lock file found in {}",
                    requested_lockfile_path.display()
                );
```

Also, I don't think we want to emit the `no Cargo.lock file published in ...` on line 194, so make then conditioned on the existence of `requested_lockfile_path`, like

```rust
if gctx.locked() {
    if let Some(requested_lockfile_path) = ws.requested_lockfile_path() {
        // error
    } else if !ws.root().join("Cargo.lock").exists() {
        // warn
    }
} 
```

Reply:
Apologies, I believe it was left from when the logic was to copy it. Fixed! Thank you 🙏🏼